### ISSUE TITLE
Wayland System Tray

### ISSUE THREAD

**Issue by xensurve** [ebruck/radiotray-ng#216]
`(radiotray-ng:2681): Gtk-CRITICAL **: 14:23:44.987: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
`

OS : ARCH
DESKTOP: PLASMA , WAYLAND
RADIOTRAY : AUR

I have installed lib32-libappindicator-gtk3 and libappindicator-gtk3 but it stil wont show in the system tray , works fine in X11

**Comment by ebruck** [comment]
Have you tried any other appindicator app under KDE? 
I know that it works to some degree, but it isn't as nice as on Gnome. 
For me radiotray-ng works fine under Wayland/Gnome.

**Comment by idovitz** [comment]
Same here, no tray icon in gnome3, works fine under plasma6/wayland..

**Comment by xensurve** [comment]
here's a work around

GDK_BACKEND=x11 radiotray-ng

**Comment by ebruck** [comment]
> Same here, no tray icon in gnome3, works fine under plasma6/wayland..

Do you have this extension installed for gnome3?
https://extensions.gnome.org/extension/615/appindicator-support/

**Comment by idovitz** [comment]
Yep, installed it. Only show tray icon with xensurves workarround.

**Comment by ebruck** [comment]
@idovitz if you run this command what does it print?

`GD_BACKEND=help radiotray-ng`

**Comment by idovitz** [comment]
Supported GDK backends: wayland x11 broadway

(radiotray-ng:4311): Gtk-CRITICAL **: 18:09:43.156: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed

**Comment by idovitz** [comment]
Supported GDK backends: wayland x11 broadway

(radiotray-ng:4311): Gtk-CRITICAL **: 18:09:43.156: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed

**Comment by SpyPower** [comment]
the .desktop file needs to be altered.

```
$ cat /usr/share/applications/radiotray-ng.desktop
[Desktop Entry]
Categories=AudioVideo;Audio;Player;GTK;
Comment=Internet Radio Player for Linux
Exec=/usr/bin/env GDK_BACKEND=x11 /usr/bin/radiotray-ng
Hidden=false
Icon=/usr/share/icons/hicolor/256x256/apps/radiotray-ng-notification.png
Name=Radiotray-NG
StartupNotify=false
Terminal=false
Type=Application
```

**Comment by human1ty** [comment]
Still an issue
`GNOME Shell 48.1` + [gnome-shell-extension-appindicator](https://github.com/ubuntu/gnome-shell-extension-appindicator)
`(radiotray-ng:225616): Gtk-CRITICAL **: 20:14:05.094: gtk_widget_get_parent: assertion 'GTK_IS_WIDGET (widget)' failed`
The app hangs
